### ISSUE TITLE
manual/javaGuide/main/http/JavaBodyParsers.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#952]
- https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/javaGuide/main/http/JavaBodyParsers.md
- diffs between [2.2.0/manual/javaGuide/main/http/JavaBodyParsers.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/javaGuide/main/http/JavaBodyParsers.md) and [2.3.5/manual/javaGuide/main/http/JavaBodyParsers.md](https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/javaGuide/main/http/JavaBodyParsers.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/javaGuide/main/http/JavaBodyParsers.md  2013-09-19 10:11:22.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/javaGuide/main/http/JavaBodyParsers.md 2014-08-12 10:21:26.000000000 +0900
@@ -1,49 +1,33 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Body parsers

 ## What is a body parser?

 An HTTP request (at least for those using the POST and PUT operations) contains a body. This body can be formatted with any format specified in the Content-Type header. A **body parser** transforms this request body into a Java value. 

-> **Note:** You can't write `BodyParser` implementation directly using Java. Because a Play `BodyParser` must handle the body content incrementaly using an `Iteratee[Array[Byte], A]` it must be implemented in Scala.
+> **Note:** You can't write `BodyParser` implementation directly using Java. Because a Play `BodyParser` must handle the body content incrementally using an `Iteratee[Array[Byte], A]` it must be implemented in Scala.
 >
 > However Play provides default `BodyParser`s that should fit most use cases (parsing Json, Xml, Text, uploading files). And you can reuse these default parsers to create your own directly in Java; for example you can provide an RDF parsers based on the Text one.

 ## The `BodyParser` Java API

+When working with request bodies, ensure that have the following imports in your controller:
+
+@[imports](code/javaguide/http/JavaBodyParsers.java)
+
 In the Java API, all body parsers must generate a `play.mvc.Http.RequestBody` value. This value computed by the body parser can then be retrieved via `request().body()`:

-'''java
-public static Result index() {
-  RequestBody body = request().body();
-  return ok("Got body: " + body);
-}
-'''
+@[request-body](code/javaguide/http/JavaBodyParsers.java)

 You can specify the `BodyParser` to use for a particular action using the `@BodyParser.Of` annotation:

-'''java
-@BodyParser.Of(BodyParser.Json.class)
-public static Result index() {
-  RequestBody body = request().body();
-  return ok("Got json: " + body.asJson());
-}
-'''
+@[particular-body-parser](code/javaguide/http/JavaBodyParsers.java)

 ## The `Http.RequestBody` API

 As we just said all body parsers in the Java API will give you a `play.mvc.Http.RequestBody` value. From this body object you can retrieve the request body content in the most appropriate Java type.

-> **Note:** The `RequestBody` methods like `asText()` or `asJson()` will return null if the parser used to compute this request body doesn't support this content type. For example in an action method annotated with `@BodyParser.Of(BodyParser.Json.class)`, calling `asXml()` on the generated body will retun null.
-
-Some parsers can provide a most specific type than `Http.RequestBody` (ie. a subclass of `Http.RequestBody`). You can automatically cast the request body into another type using the `as(...)` helper method:
-
-'''java
-@BodyParser.Of(BodyLengthParser.class)
-public static Result index() {
-  BodyLength body = request().body().as(BodyLength.class);
-  ok("Request body length: " + body.getLength());
-}
-'''
+> **Note:** The `RequestBody` methods like `asText()` or `asJson()` will return `null` if the parser used to compute this request body doesn't support this content type. For example in an action method annotated with `@BodyParser.Of(BodyParser.Json.class)`, calling `asXml()` on the generated body will return `null`.

 ## Default body parser: AnyContent

@@ -56,44 +40,20 @@
 - **multipart/form-data**: `Http.MultipartFormData`, accessible via `asMultipartFormData()`
 - Any other content type: `Http.RawBuffer`, accessible via `asRaw()`

-Example:
+For example:

-'''java
-public static Result save() {
-  RequestBody body = request().body();
-  String textBody = body.asText();
-  
-  if(textBody != null) {
-    ok("Got: " + text);
-  } else {
-    badRequest("Expecting text/plain request body");
-  }
-}
-'''
+@[default-parser](code/javaguide/http/JavaBodyParsers.java)

 ## Max content length

 Text based body parsers (such as **text**, **json**, **xml** or **formUrlEncoded**) use a max content length because they have to load all the content into memory. 

-There is a default content length (the default is 100KB). 
-
-> **Tip:** The default content size can be defined in `application.conf`:
-> 
-> `parsers.text.maxLength=128K`
+There is a default maximum content length of 100KB.  It can be overridden by specifying the `parsers.text.maxLength` property in `application.conf`:

+    parsers.text.maxLength=128K

 You can also specify a maximum content length via the `@BodyParser.Of` annotation:

-'''java
-// Accept only 10KB of data.
-@BodyParser.Of(value = BodyParser.Text.class, maxLength = 10 * 1024)
-public static Result index() {
-  if(request().body().isMaxSizeExceeded()) {
-    return badRequest("Too much data!");
-  } else {
-    ok("Got body: " + request().body().asText()); 
-  }
-}
-'''
+@[max-length](code/javaguide/http/JavaBodyParsers.java)

 > **Next:** [[Actions composition | JavaActionsComposition]]

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged #1126
